R/pivot_tables.R

```diff
@@ -255,32 +258,35 @@ ColocalizationScoresToAssay.CellGraphAssay <- function (
   ...
 ) {
 
-  # fetch polarization scores
-  col_table <- slot(object, name = "colocalization")
-  if (is.null(col_table)) {
-    abort("'colocalization' scores are missing from 'CellGraphAssay'")
-  }
+  coloc_matrix <- .create_spatial_metric_matrix(object,
+                                                values_from = values_from,
+                                                metric = "colocalization")
 
-  # Validate input
-  values_from <- match.arg(values_from,
-                           choices = c("pearson_z", "pearson", "pearson_mean",
-                                       "jaccard_mean", "jaccard", "jaccard_z"))
+  # Create Assay from filled matrix
+  assay <- CreateAssayObject(data = coloc_matrix)
 
-  col_scores_wide_format <- ColocalizationScoresToAssay(col_table, values_from, ...)
+  return(assay)
+}
 
-  # Make sure that the components match
-  # Create an empty matrix (all 0's)
-  tofillMat <- matrix(data = 0,
-                      nrow = nrow(col_scores_wide_format),
-                      ncol = ncol(object),
-                      dimnames = list(rownames(col_scores_wide_format), colnames(object)))
 
-  # Fill matrix where it overlaps
-  # Any missing columns will be keep 0's
-  tofillMat[, colnames(col_scores_wide_format)] <- col_scores_wide_format
+#' @rdname ColocalizationScoresToAssay
+#' @method ColocalizationScoresToAssay CellGraphAssay5
+#'
+#' @export
+#'
+ColocalizationScoresToAssay.CellGraphAssay5 <- function (
+  object,
+  values_from = c("pearson_z", "pearson", "pearson_mean",
+                  "jaccard_mean", "jaccard", "jaccard_z"),
```

Comment:
The pearson_mean and jaccard_mean are the random means estimated from permutations. Should they really be retrievable?

Reply:
I think that's a fair point. I would much rather trim down this list! Maybe "pearson_z" and "pearson" is enough?

Reply:
I would think so, yes!